File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    k=0
    student1 = [1,2,3,4,5]
    s1 = 0
    student2 = [2,1,2,3,2,4,2,5]
    s2 = 0
    student3 = [3,3,1,1,2,2,4,4,5,5]
    s3 = 0
    c1=0
    c2=0
    c3=0
    for i in range(len(answers)):
        s1 = i%5
        s2 = i%8
        s3 = i%10
        
        if student1[s1] == answers[i]:
            c1 +=1
        if student2[s2] == answers[i]:
            c2 +=1
        if student3[s3] == answers[i]:
            c3 +=1
    k = max(c1,c2,c3)
    answer = []
    if k == c1:
        answer.append(1)
    if k == c2:
        answer.append(2)
    if k == c3:
        answer.append(3)
        
    
    return answer

File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        
        int[] s1 = new int[]{1,2,3,4,5};
        int[] s2 = new int[]{2,1,2,3,2,4,2,5};
        int[] s3 = new int[]{3,3,1,1,2,2,4,4,5,5};
        int c1=0;
        int c2=0;
        int c3=0;
        for(int i=0; i<answers.length; i++){
            int ss1 = i%5;
            int ss2 = i%8;
            int ss3 = i%10;
            if(s1[ss1] == answers[i]){
                c1++;
            }
            if(s2[ss2]==answers[i]){
                c2++;
            }
            if(s3[ss3]==answers[i]){
                c3++;
            }
        
        }
        ArrayList<Integer> list = new ArrayList<>();
        int max = Math.max(c1,Math.max(c3,c2));
        if(max == c1) list.add(1);
        if(max == c2) list.add(2);
        if(max == c3) list.add(3);
        int[] answer = new int[list.size()];
        for(int i=0; i<list.size(); i++){
            answer[i] = list.get(i);
        }
        return answer;
    }
}